### PR TITLE
Add more unit tests for page components/controls

### PR DESCRIPTION
**Pull request by huong-li-nguyen**
## Description
- Update docs to show how to use Button as a ControlType via custom component approach
- Add unit tests for valid components/controls

Follow-up PR to this: https://github.com/mckinsey/vizro/pull/88

## Checklist

- [x] I have not referenced individuals, products or companies in any commits, directly or indirectly
- [x] I have not added data or restricted code in any commits, directly or indirectly
- [x] I have updated the docstring of any public function/class/model changed
- [x] I have added the PR number to the change description in the changelog fragment, e.g. `Enable feature XXX ([#1](https://github.com/mckinsey/vizro/pull/1))` (if applicable)
- [x] I have added tests to cover my changes (if applicable)

## Types of changes

- [x] Docs/refactoring (non-breaking change which improves codebase)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

  - I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
  - I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
  - I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

File: vizro-core/docs/pages/user_guides/components.md [hunk 1]
```diff
@@ -762,3 +762,19 @@ In the below example we show how to configure a button to export the filtered da
         [![Button]][Button]
 
     [Button]: ../../assets/user_guides/components/button.png
+
+The [`Button`][vizro.models.Button] component is currently reserved to be used inside the right-side of the dashboard (output side).
+However, there might be use cases where one wants to place the `Button` on the left-side with the controls.
```
Comment (maxschulz-COL) [inline vizro-core/docs/pages/user_guides/components.md:767]:
```suggestion
The [`Button`][vizro.models.Button] component is currently reserved to be used inside the main panel of the dashboard.
However, there might be use cases where one would like to place the `Button` inside the control panel (left-side) with the other controls.
```

File: vizro-core/docs/pages/user_guides/components.md [hunk 2]
```diff
@@ -762,3 +762,19 @@ In the below example we show how to configure a button to export the filtered da
...
+
+In this case, simply follow the user-guide outlined for [custom components](custom_components.md) and manually add the `Button` as a control type
+by running the following lines before your dashboard configurations:
```
Comment (maxschulz-COL) [inline vizro-core/docs/pages/user_guides/components.md:770]:
```suggestion
In this case, simply follow the user-guide outlined for [custom components](custom_components.md) and manually add the `Button` as a valid type to the `controls` argument by running the following lines before your dashboard configurations:
```

**Review by maxschulz-COL — APPROVED** [review]
Lgtm, just 2 suggestions to make the text flow a little more

**Comment by huong-li-nguyen** [comment]
> Lgtm, just 2 suggestions to make the text flow a little more

I like it - thanks! 📝

**Review by Joseph-Perkins — APPROVED** [review]

**Pull request merged**